src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlVirtualDocumentFactory.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Composition;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServerClient.Razor
+{
+    [Shared]
+    [Export(typeof(VirtualDocumentFactory))]
+    internal class HtmlVirtualDocumentFactory : VirtualDocumentFactory
```

Comment:
>HtmlVirtualDocumentFactory [](start = 19, length = 26)

Nit: seems 90% the same as the CSharp one, not sure if some sort of code re-use would make sense (base class or generic class), though the code is simple and I like the readability of it.

Reply:
For now it's 90% the same. When we add more pieces to each of these factories (i.e. C# adding the buffer to the workspace) they'll probably diverge significantly